File: src/main/java/com/jgroup/farmers_market/model/api/JwtResponse.java
package com.jgroup.farmers_market.model.api;

import lombok.Data;

import java.util.List;

@Data
public class JwtResponse {
    private String accessToken;
    private String tokenType = "Bearer";
    private String username;
    private List<String> roles;

    public JwtResponse(String accessToken, String username, List<String> roles) {
        this.accessToken = accessToken;
        this.username = username;
        this.roles = roles;
    }
}

File: mobile/lib/auth/jwt_response.dart
class JwtResponse {
  final String accessToken;
  final String username;
  final List<String> roles;

  JwtResponse({
    required this.accessToken,
    required this.username,
    required this.roles,
  });

  factory JwtResponse.fromJson(Map<String, dynamic> json) {
    return JwtResponse(
      accessToken: json['accessToken'],
      username: json['username'],
      roles: List<String>.from(json['roles']),
    );
  }
}
